### ISSUE TITLE
Q: Torch memory usage vs Tensorflow

### ISSUE THREAD

**Issue by teaglin** [keras-team/keras#21870]
Hi, 

For reference I am using torch=2.7.0+cu128 and tensorflow=2.20.0 both using CUDA/GPU enabled.

Let's say I attempt to train a model with batch size of 96. Tensorflow handles it no problem. If I try to train the same model with pytorch I can barely do a batch size of 24 with out it crashing due to memory.

Is there something else going on that causes this discrepancy, or is pytorch more of a memory hog? Should I update my pytorch version?

**Comment by dhantule** [comment]
Hi @teaglin, thanks for reporting this. 
please provide reproducible code to reproduce this issue

**Comment by maitry63** [comment]
Hi @teaglin , PyTorch doesn’t inherently use more memory than TensorFlow, but defaults like activations caching, autograd, and DataLoader behavior cause higher memory usage. 

To train large batch sizes in PyTorch: Enable mixed precision Use gradient checkpointing Apply gradient accumulation optimize DataLoader and cuDNN settings then PyTorch can often match or even outperform TensorFlow in memory efficiency.

TensorFlow also performs more graph-level fusion that reduces intermediate activations. Differences in data pipeline behaviour (PyTorch DataLoader vs TF tf.data) can also increase PyTorch GPU usage. Updating to the latest stable PyTorch CUDA build is recommended. 
Try to upgrade to latest PyTorch version & it should run faster as TensorFlow.

For reference check:
TensorFlow GPU memory [docs](https://www.google.com/url?q=https%3A%2F%2Fwww.tensorflow.org%2Fguide%2Fgpu%23limiting_gpu_memory_growth).

**Comment by teaglin** [comment]
@maitry63 Thanks, I will try those out.

**Comment by maitry63** [comment]
@teaglin Feel free to close this issue, if it gets resolved from your end.
